packages/block-editor/src/hooks/layout.js

```diff
@@ -212,9 +255,16 @@ export const withLayoutStyles = createHigherOrderComponent(
 		const usedLayout = layout?.inherit
 			? defaultThemeLayout
 			: layout || defaultBlockLayout || {};
-		const className = classnames( props?.className, {
-			[ `wp-container-${ id }` ]: shouldRenderLayoutStyles,
-		} );
+		const layoutClasses = shouldRenderLayoutStyles
+			? getLayoutClasses( attributes )
+			: null;
```

Comment:
@andrewserong there were a few different ways of handling this, but this seemed the clearest. I thought about passing `shouldRenderLayoutStyles` into `getLayoutClasses` and returning the empty array if false, or `layoutClasses = shouldRenderLayoutStyles && getLayoutClasses( attributes );` is slightly more succinct and works, but `&&` seems a little hacky in variable assignment to me - sort of ok in React rendering.

Reply:
It's funny with these little things, there's so many different ways to do it 😄. This way looks good to me!